include/aws/io/pem.h

```diff
@@ -0,0 +1,95 @@
+#ifndef AWS_IO_PEM_H
+#define AWS_IO_PEM_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/io/io.h>
+
+AWS_EXTERN_C_BEGIN
+
+enum aws_pem_object_type {
+    AWS_PEM_TYPE_UNKNOWN = 0,
+    AWS_PEM_TYPE_X509_OLD,
+    AWS_PEM_TYPE_X509,
```

Comment:
is there a more specific name than "old"? or is this purposefully vague, lumping together several types we no longer support? Add comment if this is something worth explaining

Reply:
its trying to stay close to openssl naming for pem types. i can add a comment on the expected type string for each in file